provisioner/puppet-masterless/provisioner.go

```diff
@@ -259,7 +259,13 @@ func (p *Provisioner) uploadManifests(ui packer.Ui, comm packer.Communicator) (s
 	}
 	defer f.Close()
 
-	manifestFilename := filepath.Base(p.config.ManifestFile)
+	manifestFilename := p.config.ManifestFile
+	if fi, err := os.Stat(p.config.ManifestFile); err != nil {
+		return "", fmt.Errorf("Error inspecting manifest file: %s", err)
+	} else if !fi.IsDir() {
```

Comment:
Or If we want to discourage this going forward we could log a warning when we catch the overloaded directory use case. That suffices in place of a comment imo.